### PR TITLE
Mouse wheel input refactor

### PR DESCRIPTION
**Pull request by ZehMatt**
The goal should be following:
input (4)

This PR also fixes the odd behavior that it accumulates a delta which requires to first undo the scroll before it would actually scroll.

### CODE REVIEW

**Review by duncanspumpkin — APPROVED** [review]

**Comment by AaronVanGeffen** [comment]
> This PR also fixes the odd behavior that it accumulates a delta which requires to first undo the scroll before it would actually scroll.

Could you add a changelog entry for this?

File: src/OpenLoco/src/Input/MouseInput.cpp [hunk 1]
```diff
@@ -2292,4 +2299,70 @@ namespace OpenLoco::Input
...
+                    main->viewportRotateRight();
+                }
+                else if (wheel < 0)
+                {
+                    main->viewportRotateLeft();
+                }
+                TownManager::updateLabels();
+                StationManager::updateLabels();
+                Windows::MapWindow::centerOnViewPoint();
+            }
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Input/MouseInput.cpp:2356]: Could this be better delegated to the window manager as well?

Comment (ZehMatt) [reply src/OpenLoco/src/Input/MouseInput.cpp:2356]: Most likely, not for this PR, been meaning to create a chart first to see where is what and this PR is rather a side effect of me wanting to get the mouse wheel working correctly with the bonus of cleanup.

**Review by AaronVanGeffen — CHANGES_REQUESTED** [review]
Some bad news. This PR breaks scrolling using my touchpad entirely.

**Review by AaronVanGeffen — APPROVED** [review]
All good now! Thanks for addressing the issue so quickly.

**Pull request merged**